### ISSUE TITLE
Subscription to On(Location) stops firing after sometime

### ISSUE THREAD

**Issue by 8bhsolutions** [mauron85/cordova-plugin-background-geolocation#612]
## Your Environment

* Plugin version: Using a fork from Master (Apr 11, 2019)
* Platform: Android
* OS version: 9.0
* Device manufacturer and model: Samsung S9+

* Running in Simulator:
* Cordova version (`cordova -v`):
* Cordova platform version (`cordova platform ls`):
* Plugin configuration options: 
```
const config: BackgroundGeolocationConfig = {
            desiredAccuracy: 0,
            stationaryRadius: this.updateDistance / 10,
            distanceFilter: this.updateDistance,
            interval: this.updateInterval,
            fastestInterval: this.updateInterval,
            startForeground: true,
            debug: false,
        };
```
* Link to your project:
* Other info:
Ionic:

   Ionic CLI          : 5.2.3
   Ionic Framework    : ionic-angular 3.9.2
   @ionic/app-scripts : 3.2.4

Cordova:

   Cordova CLI       : 9.0.0 (cordova-lib@9.0.1)
   Cordova Platforms : android 8.0.0
   Cordova Plugins   : cordova-plugin-ionic-keyboard 2.1.3, cordova-plugin-ionic-webview 3.1.2, (and 28 other plugins)

Utility:

   cordova-res : not installed
   native-run  : 0.2.8

System:

   Android SDK Tools : 26.1.1 
   NodeJS            : v10.16.0 
   npm               : 6.9.0
   OS                : Windows 10

## Context
Using the plugin to track gps location.

I'm using [Background Mode](https://github.com/katzer/cordova-plugin-background-mode) and [powermanagement](https://github.com/boltex/cordova-plugin-powermanagement) plugins to keep my app alive in background.
The app is alive, however, the GPS location stops updating after some time (sometime 30ish minutes, sometimes more than 2 hours).
The Background Geolocation notification is still on.

Investigating into BackgroundGeolocationLogEntry shows that location change events seems to have stopped firing.
The event entries continue with lots of Provider gps status changed: 1 
and other common entries, however there are no Location change log entries.

If I stop tracking, and restart. Everything works again.

My app level logging did not catch any errors. However, interestingly, the last location BackgroundGeolocationResponse had id: undefined.

## Expected Behavior
Background tracking should continue returning location updates
Or throw a error so the application can handle with error scenarios

## Actual Behavior
Background tracking silently fails

## Steps to Reproduce
Don't know exactly how to reproduce.

1. Call backgroundGeolocation.Start()
2. Let user use app. Including multi-task with other apps
3. After sometime, it stops.
4.

## Context

```

this.backgroundGeolocation.configure(config)
            .then(() => {
                this.locationUpdateSubscription = this.backgroundGeolocation.on(BackgroundGeolocationEvents.location)
                    .subscribe((location: BackgroundGeolocationResponse) => {
                        this.log.log(`BackgroundGeolocation - ${location.id}, ${location.time}:  ${location.latitude}, ${location.longitude}`);

                        this.addLocationToSession(location);

                        // Update inside of Angular zone
                        this.zone.run(() => {
                            this.lat = location.latitude;
                            this.lng = location.longitude;
                            this.updateTime = new Date(location.time);
                        });

                        // IMPORTANT:  You must execute the finish method here to inform the native plugin that you're finished,
                        // and the background-task may be completed.  You must do this regardless if your operations are successful or not.
                        // IF YOU DON'T, ios will CRASH YOUR APP for spending too much time in the background.
                        this.backgroundGeolocation.finish(); // FOR IOS ONLY
                    }, (err) => {
                        this.log.err("Background geolocation tracking issue.", err);
                    }, () => {
                        this.log.log("Background geolocation tracking ended successfully");
                    });
            });

this.backgroundGeolocation.start();

```

## Debug logs
unable to get debug logs working with instructions.
Built a page in my app to browse the logs.
Screenshot attached
![Screenshot_20190805-183140_Baxta](https://user-images.githubusercontent.com/48874658/62450442-aceec600-b7af-11e9-98a3-93157b7b37ef.jpg)

Screenshot shows just after last location update

**Comment by 8bhsolutions** [comment]
**Additional information**
Seems that before plugin starts to fail, it receives multiple duplicate events.
For my test below, the last successful BGLocation event was coupled with 5 LocationUpdate events which are milliseconds apart.
However, this shouldn't happen in my setup, because I am expecting only 1 LocationUpdate event per minute.
Something internal to the plugin/filter code is failing.

The location update id being null, seems to be a red herring, because other tests failures have values for the location id.

**Test setup**
I ran a session with distanceFilter: 1, and interval:60000. So effectively a location update every 60 seconds.
The session ran for 27+ hours without a problem, and location updates roughly every minute. No code change from any of the other test sessions. And then suddenly no more location updates.
I left it running for another 2 hours after it stopped providing location updates.

The plugin event logs are the same after plugin stops working. 
Plenty of Provider gps status changed: 1, but no Location update events.

The 2nd last LocationUpdate event had 5 multiple events milliseconds apart, that had same values, only the **et** value was milliseconds apart.

What I've noticed is that the previous session with failure had the same odd occurance.
It seems that the the multiple LocationUpdate events, milliseconds apart, causes some code to fail, and the service does not recover
